Scripts/Python/xAgeSDLBoolActivatorComboSet.py

```diff
@@ -194,9 +210,20 @@ def _RespPushComplete(self, events):
         self._HitActId = None
 
     def _TriggerButton(self, actId):
+        self._AddToAttempt(actId)
+
         if self._solved:
             PtDebugPrint("xAgeSDLBoolActComboSet._TriggerButton():\tYou just unsolved it, moron.", level=kWarningLevel)
             self._solved = False
+            self._numCorrect = 0
+            self._attemptCombo = [actId]
+            return
+
+        if allowSlidingSolution.value:
+            attempt = self._attemptCombo
+            self._numCorrect = next((len(attempt) - i for i in range(len(attempt)) if self._IsAttemptCorrectAtIndex(i)), 0)
```

Comment:
This looks a little weird to me. I think I could use an explanation either here on in the comments about the desired process.

Reply:
I'll try to put a comment in the code, but basically this is looping through partial slices of the attempt array and comparing against the *start* of the combination array (up to the length of the attempt so far). For example, an attempt [3, 2, 1] for a solution of [1, 2, 3, 4, 5] would do the following comparison:
First -- index 0: [3, 2, 1] == [1, 2, 3]
Second -- index 1: [2, 1] == [1, 2]
Third -- index 2: [1] == [1]

and then return 1 (length of 3 minus index value of 2) as the number of correct values in the attempt. If there is no match, it uses the default value of 0 returned from the `next` iterator.